tests/Doctrine/Tests/ORM/Functional/Ticket/GH6464Test.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+namespace Doctrine\Tests\ORM\Functional\Ticket;
+
+use Doctrine\Tests\OrmFunctionalTestCase;
+
+/**
+ * @group GH-6464
+ */
+class GH6464Test extends OrmFunctionalTestCase
+{
+    /**
+     * {@inheritDoc}
+     */
+    protected function setUp()
+    {
+        parent::setUp();
+
+        $this->_schemaTool->createSchema(
+            [
+                $this->_em->getClassMetadata(GH6464Post::class),
+                $this->_em->getClassMetadata(GH6464User::class),
+                $this->_em->getClassMetadata(GH6464Author::class),
+            ]
+        );
+    }
+
+    /**
+     * Verifies that SqlWalker generates valid SQL for an INNER JOIN to CTI table
+     *
+     * SqlWalker needs to generate nested INNER JOIN statements, otherwise there would be INNER JOIN
+     * statements without an ON clause, which are valid on e.g. MySQL but rejected by PostgreSQL.
+     */
+    public function testIssue()
+    {
+        $query = $this->_em->createQueryBuilder()
+            ->select('p', 'a')
```

Comment:
This sort of `select` is invalid - removing `a` from the selection, as you should *NEVER* fetch-join results over an association that was filtered.